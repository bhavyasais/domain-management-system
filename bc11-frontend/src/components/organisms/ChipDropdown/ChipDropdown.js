import { Typography, withStyles } from '@material-ui/core';
import React, { useCallback, useState } from 'react';
import KeyboardArrowDownIcon from '@material-ui/icons/KeyboardArrowDown';
import clsx from 'clsx';
import PropTypes from 'prop-types';
import Chip from '../../atoms/Chip/Chip';
import { TRUSTGROUP_COLORS, RELATIONSHIP_COLORS } from '../../../utils/Strings';

const styles = () => ({
  DropDownContainer: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    padding: '8px',
    border: '1px solid #D3D3D3',
    backgroundColor: '#FFFFFF',
    marginTop: '4px',
  },
  dropDownHeader: {
    alignItems: 'center',
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingLeft: '8px',
    paddingBottom: '6px',
  },
  defaultStyle: {
    width: '180px',
  },
  defaultItemStyle: {
    marginBottom: '8px',
    '&:hover': {
      backgroundColor: '#f6f6f6',
    },
  },
});

const ChipDropdown = ({
  placeholder,
  list,
  listOpen,
  onDropDownChange,
  customStyle,
  placeholderStyle,
  classes,
  selectedValue,
  dropDownContainerStyle,
  itemStyle,
}) => {
  const [isOpen, setisOpen] = useState(listOpen);
  const [value, setValue] = useState(selectedValue);

  const onToggleList = () => {
    setisOpen(!isOpen);
  };

  const handleDropDownChange = useCallback(
    (index) => () => {
      setValue(list[index]?.groupName);
      onDropDownChange?.(list[index], index);
      onToggleList();
    },
    [list, onDropDownChange, onToggleList],
  );
  return (
    <div className={clsx(classes.defaultStyle, customStyle)}>
      <div className={classes.dropDownHeader} onClick={() => onToggleList?.()}>
        {value ? (
          <Chip
            label={value}
            backgroundColor={TRUSTGROUP_COLORS[value] ?? RELATIONSHIP_COLORS[value] ?? '#eeeeee'}
            color="#1c1c21"
          />
        ) : (
          <Typography className={placeholderStyle}>{placeholder}</Typography>
        )}
        <KeyboardArrowDownIcon className={classes.menuIcon} />
      </div>
      {isOpen && (
        <div className={clsx(classes.DropDownContainer, dropDownContainerStyle)}>
          {list?.map((item, index) => (
            <div
              onClick={handleDropDownChange(index)}
              key={index}
              className={clsx(classes.defaultItemStyle, itemStyle)}
            >
              <Chip
                label={item?.groupName}
                backgroundColor={
                  TRUSTGROUP_COLORS[item?.groupName] ??
                  RELATIONSHIP_COLORS[item?.groupName] ??
                  '#eeeeee'
                }
                color="#1c1c21"
              />
            </div>
          ))}
        </div>
      )}
    </div>
  );
};

ChipDropdown.propTypes = {
  placeholder: PropTypes.string,
  selectedValue: PropTypes.string,
};
ChipDropdown.defaultProps = {
  placeholder: 'Find an option',
  selectedValue: null,
  placeholderStyle: null,
  customStyle: null,
  dropDownContainerStyle: null,
  listOpen: false,
};

export default withStyles(styles)(ChipDropdown);
