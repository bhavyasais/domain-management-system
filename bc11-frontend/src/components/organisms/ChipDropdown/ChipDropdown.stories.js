import React from "react";
import ChipDropdown from "./ChipDropdown";
import { UpdateTrustGroups } from "../../../utils/helper";

export default {
  title: "Molecules/ComponentDropDown",
};

const trustGroups = ["Top 100", "Top 1000", "Top 3000", "Do not trust"];
const formattedTrustGroups = UpdateTrustGroups(trustGroups);

const onChange = () => {};

export const ComponentDropDown = () => {
  return (
    <ChipDropdown list={formattedTrustGroups} onDropDownChange={onChange} />
  );
};
