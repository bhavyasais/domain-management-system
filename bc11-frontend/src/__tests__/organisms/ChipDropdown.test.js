import React from 'react';
import renderer from 'react-test-renderer';
import ChipDropdown from '../../components/organisms/ChipDropdown/ChipDropdown';
import { mount } from 'enzyme';
import { act } from 'react-dom/test-utils';

describe('ChipDropdown', () => {
  let component;
  const handleClick = jest.fn();
  beforeEach(() => {
    act(() => {
      component = mount(<ChipDropdown onClick={handleClick} listOpen={false} />);
    });
  });
  it('Renders correctly', () => {
    let wrapper = renderer.create(<ChipDropdown />);
    expect(wrapper).toMatchSnapshot();
  });

  it('clicks correctly', () => {
    let wrapper = mount(<ChipDropdown onClick={handleClick} listOpen={false} />);
    wrapper.find('div').at(1).props().onClick();
    expect(wrapper).toMatchSnapshot();
  });
});
