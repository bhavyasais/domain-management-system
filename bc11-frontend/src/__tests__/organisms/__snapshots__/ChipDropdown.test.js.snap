// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChipDropdown Renders correctly 1`] = `
<div
  className="ChipDropdown-defaultStyle-3"
>
  <div
    className="ChipDropdown-dropDownHeader-2"
    onClick={[Function]}
  >
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      Find an option
    </p>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
      />
    </svg>
  </div>
</div>
`;

exports[`ChipDropdown clicks correctly 1`] = `ReactWrapper {}`;
